Sources/XcodeGenKit/SchemeGenerator.swift

```diff
@@ -359,6 +359,24 @@ public class SchemeGenerator {
         )
     }
 
+    public static func generateSchemeManagement(project: Project) -> XCSchemeManagement {
+        let userStateSchemes = project.targets.map { target -> XCSchemeManagement.UserStateScheme in
+            XCSchemeManagement.UserStateScheme(
+                name: target.name + ".xcscheme",
```

Comment:
This won't work with `configVariants`, as the name isn't just the target name. This logic is container with `SchemeGenerator.generateSchemes` Maybe it's safer to loop over the generated schemes. Perhaps the `SchemeGenerator.generateSchemes` function could be split into `generateSchemes() throws -> [Scheme]` and `generateXCSchemes() throws -> [XCScheme]`. This could then use the former and get both all the schemes, and their proper names

Reply:
> Maybe it's safer to loop over the generated schemes.

You mean the generated `Schema`, not `XCScheme` right? 

> Perhaps the SchemeGenerator.generateSchemes function could be split 

So something like this? 

    public func generateSchemes(project: Project) throws -> [Scheme] {
        // generating list of all schemas
    }

    public func generateXCSchemes(schemes: [Scheme]) throws -> [XCScheme] {
        // call generateScheme
    }

The only problem is, that the `generateScheme` function needs the optional `ProjectTarget` as well.. 

We could do something like this though: 

    static public func generateSchemes(project: Project) throws -> [(Scheme, ProjectTarget?)] {
        // generating list of all schemas
    }

    public func generateXCSchemes(schemes: [(Scheme, ProjectTarget?)]) throws -> [XCScheme] {
        try schemes.map { (schema, projectTarget) in
            try generateScheme(schema, for: projectTarget)
        }
    }

But this has another problem. `generateSchemeManagement` is currently not called in the `ProjectGenerator` but in the `GenerateCommand`. Which means that we need to call `generateSchemes` twice, which can't be good for performance.

Reply:
This has been addressed. The `generateSchemes` method now [returns a tuple of the shares schemes, the user schemes and the management](https://github.com/yonaskolb/XcodeGen/pull/1142/files#diff-5353e75c7700fffa19ecb24d2f9d5cae41af5fc25ef10e57c2219696c933e5f5R43-R47) which can then be used by the `ProjectGenerator` so it can be [written to the right files](https://github.com/yonaskolb/XcodeGen/pull/1142/files#diff-5b091a15a76b16e0bcb40ce6236420161ffd95493c4cb8414a81ec2e3f89f129R27-R36)